src/change_stream.ts

```diff
@@ -219,69 +300,78 @@ export class ChangeStream extends EventEmitter {
    * @function
    * @param {Writable} destination The destination for writing data
    * @param {object} [options] {@link https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options|Pipe options}
-   * @returns {null}
    */
-  pipe(destination: any, options?: object): null {
+  pipe(destination: Writable, options?: PipeOptions): Writable {
     if (!this.pipeDestinations) {
       this.pipeDestinations = [];
     }
     this.pipeDestinations.push(destination);
+    if (!this.cursor) {
+      throw new MongoError('ChangeStream has no cursor, unable to pipe');
+    }
     return this.cursor.pipe(destination, options);
   }
 
   /**
    * This method will remove the hooks set up for a previous pipe() call.
    *
    * @param {Writable} [destination] The destination for writing data
-   * @returns {null}
    */
-  unpipe(destination?: any): null {
-    if (this.pipeDestinations && this.pipeDestinations.indexOf(destination) > -1) {
-      this.pipeDestinations.splice(this.pipeDestinations.indexOf(destination), 1);
+  unpipe(destination?: Writable): ChangeStreamCursor {
+    const destinationIndex = destination ? this.pipeDestinations.indexOf(destination) : -1;
+    if (this.pipeDestinations && destinationIndex > -1) {
+      this.pipeDestinations.splice(destinationIndex, 1);
+    }
+    if (!this.cursor) {
```

Comment:
Each error message is different and I can't _pipe_ that through to the getter/setter, also I'd make the maybeCursor private but its used outside of the class so it seems like it would be confusing which one to use rather than just having the checks.

Reply:
This makes sense, this trick may be useful elsewhere.